programs/application/src/lib.rs

```diff
@@ -15,8 +19,16 @@ pub mod application {
         ctx: Context<Initialize>,
         _job_ad_id: String,
         _application_id: String,
+        _general_bump: u8,
         max_allowed_stake: u32
     ) -> Result<()> {
+
+        let general_parameters = &mut ctx.accounts.general_account;
+
+        if general_parameters.authority != ctx.accounts.authority.key() {
```

Comment:
shouldn't this be caught on the macro level?

Reply:
lemme try if the has_one property works

Reply:
i have added a constraint like this  

> constraint = authority.key() == general_account.authority